testing_app/test/favorites_test.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+import 'package:testing_app/models/favorites.dart';
+import 'package:testing_app/screens/favorites.dart';
+
+Favorites favoritesList;
+
+Widget createFavoritesScreen() => ChangeNotifierProvider<Favorites>(
+      create: (context) {
+        favoritesList = Favorites();
+        return favoritesList;
+      },
+      child: MaterialApp(
+        home: FavoritesPage(),
+      ),
+    );
+
+void addRandomItems() {
+  for (var i = 0; i < 10; i += 2) {
+    favoritesList.add(Random().nextInt(50));
+  }
+}
+
+void main() {
+  group('Favorites Page Widget Tests', () {
+    testWidgets('Test if ListView shows up', (tester) async {
```

Comment:
nit: here and following `tester` => `WidgetTester tester`. Being explicit about types is in framework style guide.

Reply:
Yes, that's right. But, the linter rules used in this repo are set to follow this style. @domesticmouse might be able to tell you the reason behind it.

Reply:
Flutter samples are meant to be examples of how to use the Flutter framework, and thus are outside of the Flutter framework's style guide. 

If memory serves, we are using `pedantic` which flags this style of type specification.

Reply:
+1. We use a slightly different style guide in the samples repo. The SDK repo has some unique concerns that typical app code doesn't have to worry about.